api/core/listen_result.go

```diff
@@ -37,39 +37,37 @@ func (s *Server) ListenResult(request *ListenResultRequest, stream Core_ListenRe
 			})
 		}
 	}
-	return
+	return nil
 }
 
-func validateTaskKey(service *service.Service, taskKey string) (err error) {
+func validateTaskKey(service *service.Service, taskKey string) error {
 	if taskKey == "" || taskKey == "*" {
-		return
+		return nil
 	}
-	_, ok := service.Tasks[taskKey]
-	if ok {
-		return
+	if _, ok := service.Tasks[taskKey]; ok {
+		return nil
 	}
-	err = errors.New("Task '" + taskKey + "' doesn't exist in this service")
-	return
+	return fmt.Errorf("Task %q doesn't exist in this service", taskKey)
 }
 
-func validateOutputKey(service *service.Service, taskKey string, outputFilter string) (err error) {
+func validateOutputKey(service *service.Service, taskKey string, outputFilter string) error {
 	if outputFilter == "" || outputFilter == "*" {
-		return
+		return nil
 	}
 	if taskKey == "" {
-		err = errors.New("Cannot filter output without specifying a task")
-		return
+		return errors.New("Cannot filter output without specifying a task")
```

Comment:
let's use `fmt.Errorf` here to like that we have the same way to return errors (and later even maybe some custom error type)